docs/architecture/adr-template.md

```diff
@@ -35,6 +50,12 @@
 
 {neutral consequences}
 
+
+## Test Cases [optional]
```

Comment:
I'm not sure this makes sense. I would rather see this in an RFC and not an ADR. ADRs are not meant to be overly technical.

Reply:
It's market optional. I still think it's useful. This doesn't need to be an implementation specific test-case. It can be a feature level test-case. Instead of removing I may clarify. What do you think?

BTW: what's the difference between RFC and ADR? ADR stands for _Architecture Decision Records_ -- it already sounds technical.

Reply:
I suppose it's fine. I guess I'd like to see maybe an example or something of what this would look like or entail in an actual ADR.

> BTW: what's the difference between RFC and ADR? ADR stands for Architecture Decision Records -- it already sounds technical.

The ADR is generally more high-level and describes the context and reasoning behind a specific architecture change or introduction. It's less of a spec/technical doc, although pseudo-code is typically present. An RFC is technical in nature and more verbose. Typically, these documents are more elaborate in the implementation and design.

We don't really do both in the SDK, but I think we should. I think [this](https://engineering.atspotify.com/2020/04/14/when-should-i-write-an-architecture-decision-record/) post has a good summary.

Reply:
Test Case examples:
+ https://github.com/ethereum/EIPs/blob/master/EIPS/eip-2938.md#test-cases
+ https://github.com/ethereum/EIPs/blob/master/EIPS/eip-665.md#test-cases
+ https://github.com/ethereum/EIPs/blob/master/EIPS/eip-820.md#test-cases
+ https://github.com/ethereum/EIPs/blob/master/EIPS/eip-884.md#test-cases-and-reference-implementation

The whole idea of a template is to motivate to thinking widely. 
Also, I would like to point out, that it's not in the scope of this PR to make an example ADR for Cosmos-SDK, because we will go into a death spiral.

Reply:
> Also, I would like to point out, that it's not in the scope of this PR to make an example ADR for Cosmos-SDK, because we will go into a death spiral.

Not sure I follow.

Reply:
> We don't really do both in the SDK, but I think we should. I think this post has a good summary.
👍 

I'm not sure I agree that an RFC would necessarily be more technical than an ADR. It's more about the culture in which they're used.

I don't have a strong opinion regarding including test cases or not.

But I think we should use RFC's too based on the above article.